### ISSUE TITLE
About the model in the paper

### ISSUE THREAD

**Issue by xinruzhi** [linkenfaqiu/MTRAN#4]
Hello author, thank you very much for sharing, I can't find the MRAM module mentioned in the paper in the code you shared, and I found that the position of this residual structure summation is also a bit inconsistent with the code, I would like to ask the author if he/she has the final model code?
best wish for you！

**Comment by hczyni** [comment]
Hello, I encountered a problem while replicating this project. When the batch size is set to 1, the GPU memory gradually increases during the first round of training. Have you encountered this issue? What are your Python and PyTorch versions?best wish for you～～～

**Comment by linkenfaqiu** [comment]
> Hello, I encountered a problem while replicating this project. When the batch size is set to 1, the GPU memory gradually increases during the first round of training. Have you encountered this issue? What are your Python and PyTorch versions?best wish for you～～～

Hello~You mentioned that the network cannot run even with a batch size of 1, which could be due to the network's memory requirements exceeding the capacity of your GPU. I have provided a version with fewer parameters, and I hope it will be helpful to you. All networks are implemented using PyTorch 1.8.2 and Python 3.7.2 on the Ubuntu 18.04 system. 

Thank you for taking the time to read this project!

**Comment by linkenfaqiu** [comment]
> Hello author, thank you very much for sharing, I can't find the MRAM module mentioned in the paper in the code you shared, and I found that the position of this residual structure summation is also a bit inconsistent with the code, I would like to ask the author if he/she has the final model code? best wish for you！

Thank you for your constructive feedback! It seems that there were some errors in the version I uploaded. For instance, the PSPModule in the network code was not actually used in this paper (though you can still use it to further improve performance), and there were some issues with the attention mechanism code. I have recently fixed these errors. I truly appreciate you pointing them out, and I apologize for the delayed response!

**Comment by hczyni** [comment]
loss计算图在内存中没释放



---Original---
From: ***@***.***&gt;
Date: Thu, Nov 21, 2024 17:21 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [linkenfaqiu/MMRAN] About the model in the paper (Issue #4)




  
Hello, I encountered a problem while replicating this project. When the batch size is set to 1, the GPU memory gradually increases during the first round of training. Have you encountered this issue? What are your Python and PyTorch versions?best wish for you～～～
  
Hello~You mentioned that the network cannot run even with a batch size of 1, which could be due to the network's memory requirements exceeding the capacity of your GPU. I have provided a version with fewer parameters, and I hope it will be helpful to you. All networks are implemented using PyTorch 1.8.2 and Python 3.7.2 on the Ubuntu 18.04 system.
 
Thank you for taking the time to read this project!

**Comment by linkenfaqiu** [comment]
> loss计算图在内存中没释放
> […](#)
> ---Original--- From: ***@***.***&gt; Date: Thu, Nov 21, 2024 17:21 PM To: ***@***.***&gt;; Cc: ***@***.******@***.***&gt;; Subject: Re: [linkenfaqiu/MMRAN] About the model in the paper (Issue #4) Hello, I encountered a problem while replicating this project. When the batch size is set to 1, the GPU memory gradually increases during the first round of training. Have you encountered this issue? What are your Python and PyTorch versions?best wish for you～～～ Hello~You mentioned that the network cannot run even with a batch size of 1, which could be due to the network's memory requirements exceeding the capacity of your GPU. I have provided a version with fewer parameters, and I hope it will be helpful to you. All networks are implemented using PyTorch 1.8.2 and Python 3.7.2 on the Ubuntu 18.04 system. Thank you for taking the time to read this project! — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

您好，可以具体讲一下是在哪里存在问题吗？我在训练过程中显存占用通常保持不变。根据我的了解，默认情况下 PyTorch 在调用 loss.backward() 后，会自动释放大部分计算图。

**Comment by hczyni** [comment]
你是中国人吗，是的话加微信聊吧，邮寄用的不习惯



---Original---
From: ***@***.***&gt;
Date: Thu, Nov 21, 2024 18:45 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [linkenfaqiu/MMRAN] About the model in the paper (Issue #4)




  
loss计算图在内存中没释放
 …
 ---Original--- From: @.&gt; Date: Thu, Nov 21, 2024 17:21 PM To: @.&gt;; Cc: @.@.&gt;; Subject: Re: [linkenfaqiu/MMRAN] About the model in the paper (Issue #4) Hello, I encountered a problem while replicating this project. When the batch size is set to 1, the GPU memory gradually increases during the first round of training. Have you encountered this issue? What are your Python and PyTorch versions?best wish for you～～～ Hello~You mentioned that the network cannot run even with a batch size of 1, which could be due to the network's memory requirements exceeding the capacity of your GPU. I have provided a version with fewer parameters, and I hope it will be helpful to you. All networks are implemented using PyTorch 1.8.2 and Python 3.7.2 on the Ubuntu 18.04 system. Thank you for taking the time to read this project! — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: @.***&gt;
  
您好，可以具体讲一下是在哪里存在问题吗？我在训练过程中显存占用通常保持不变。根据我的了解，默认情况下 PyTorch 在调用 loss.backward() 后，会自动释放大部分计算图。

**Comment by hczyni** [comment]
我记得源代码这里好像没使用item，我改成这样了seg_acc += (1 - loss1.item())
